### PR TITLE
Social Notes: Add the featured image block

### PR DESCRIPTION
**Pull request by gmjuhasz**
Fixes https://github.com/Automattic/jetpack-reach/issues/244

## Proposed changes:

* Added the featured image block to the social notes template 0ee43cc59ea7ab7a6be5773b2579fc65b88b6bb0
* Hid the media picker for notes 06b9ff8ef25acb2a09abb9f62a6b73982dbca0d5

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack-reach/issues/244
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Create a new Social Note. You should see the paragraph and the featured image block below it.
* In the Jetpack sidebar you should see the share as a social post toggle. It should be disabled.
* Add a featured image in the block, the toggle should become active. Check the shares.
* Do the same with adding the image through the post sidebar.

CleanShot 2024-03-28 at 20 39 21 png

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/social-post-control/index.js [hunk 1]
```diff
@@ -18,12 +13,24 @@ const SHARE_AS_SOCIAL_POST_TEXT = __(
...
  *
- * @param {boolean} disabled -  Indicates whether the checkbox is disabled or not.
+ * @param {object} props - The properties passed to the component.
+ * @param {boolean} props.disabled -  Indicates whether the checkbox is disabled or not.
+ * @param {boolean} props.isCustomMediaAvailable - Indicates whether custom media is available or not.
  * @returns {object} The SocialPostControl Component.
  */
-export default function SocialPostControl( { disabled = false } ) {
+export default function SocialPostControl( { disabled = false, isCustomMediaAvailable = true } ) {
 	const { shouldUploadAttachedMedia, updateShouldUploadAttachedMedia } = useAttachedMedia();
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/components/social-post-control/index.js:22]: I wonder whether we should make the default for Notes to share as a social post, considering the changes we're making to the link format. Having different defaults for different post formats could be really confusing though, and someone might want to change the default. Something to think about!

I think name/label 'Share as a social post' is something we should change. It's interesting that we call it 'upload attached media' here.

**Review by pablinos — DISMISSED** [review]
This works well for me. I like the way it simplifies the interface for notes, and that you don't have to have the sidebar open. But we'll need to make sure that #36654 is merged first.

File: projects/js-packages/publicize-components/src/components/social-post-control/index.js (hunk 1 above)
Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/components/social-post-control/index.js:22]: I agree. It's little confusing.

Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/social-post-control/index.js:22]: So, which part is confusing? Should I enable it by default? I'm not sure if changing the name/label is a good idea, as it's the same functionality, just for different post types, we want the users to recognize that 🤔

Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/social-post-control/index.js:22]: >  It's interesting that we call it 'upload attached media' here.

Yeah we could refactor this component a bit in another change to make the naming more clear.

**Review by manzoorwanijk — COMMENTED** [review]
I followed the instructions but couldn't see the "Share as a social post" toggle turning ON.

https://github.com/Automattic/jetpack/assets/18226415/c7a11c40-0190-416a-8eb7-80e3db3ba736

**Comment by gmjuhasz** [comment]
> I followed the instructions but couldn't see the "Share as a social post" toggle turning ON.

I haven't made it turn on by default. Yesterday on the team call, we said to keep it consistent with the normal post type for now, I think? @spsiddarthan what do you think?

**Comment by pablinos** [comment]
> > I followed the instructions but couldn't see the "Share as a social post" toggle turning ON.
> 
> I haven't made it turn on by default. Yesterday on the team call, we said to keep it consistent with the normal post type for now, I think? @spsiddarthan what do you think?

What about turning it on by default if the feature is available? Sharing as a social post is beneficial for engagement and people on the Advanced plan should take advantage of it.

I didn't mean for this to prevent the PR from being merged. If we're still not sure, or we're turning it on for all, let's merge this and follow-up with another change.

**Comment by manzoorwanijk** [comment]
Yeah, let us merge this PR and improve that logic in a follow up PR.

**Review by manzoorwanijk — APPROVED** [review]
Let us ship it.

**Pull request merged**